presto-main/src/main/java/io/prestosql/operator/HashAggregationOperator.java

```diff
@@ -378,8 +384,14 @@ public void addInput(Page page)
                         operatorContext,
                         maxPartialMemory,
                         joinCompiler,
-                        true,
-                        useSystemMemory);
+                        () -> {
+                            aggregationBuilderMemoryContext.setBytes(((InMemoryHashAggregationBuilder) aggregationBuilder).getSizeInMemory());
```

Comment:
`InMemoryHashAggregationBuilder` instance gets a callback in its constructor, but the callback refers to that instance. Should `InMemoryHashAggregationBuilder`  use this callback in the ctor to update initial memory occupancy, this would fail.
To fix this, `UpdateMemory` should take `long bytes` parameter.

Reply:
> To fix this, UpdateMemory should take long bytes parameter.

`GroupByHash` would need to be passed `LocalMemoryContext` as an argument then. I also tried this approach, but it explodes the changes and still doesn't nicely solve problem of flipping memory from revocable to user (e.g: `GroupByHash` would need to keep `getEstimatedSize` method, but also accept `LocalMemoryContext` as arg).

Objects should either:
* expose it's memory usage via `getEstimatedSize` method (if they don't "own" the memory)
* account for memory themselves (via `LocalMemoryContext`). In this case `AggregatedMemoryContext` should be passed to constructor of such objects.

Mixing both usage patterns is confusing.